src/components/KSegmentedControl/KSegmentedControl.vue

```diff
@@ -24,102 +24,52 @@
   </div>
 </template>
 
-<script lang="ts">
-import { defineComponent, ref, PropType } from 'vue'
+<script lang="ts" setup>
+import { ref, PropType } from 'vue'
 import KButton from '@/components/KButton/KButton.vue'
+import { SegmentedControlOption } from '@/types/segmented-control'
+// need to be imported as const from separate file due to an error "defineProps are referencing locally declared variables"
```

Comment:
You can fix this by including the items in a separate `<script lang="ts">` block (no `setup`) like we are doing here: https://github.com/Kong/kongponents/blob/main/src/components/KMultiselect/KMultiselect.vue#L244

If you have the non-`setup` `script` block BEFORE the `setup` block you need to move all `import` statements into the non-`setup` block. If you have the non-`setup` block AFTER the `setup` block you only have to move the functions being used in `defineProps`.

Reply:
then I got another error with duplicated import of type

Reply:
updated